Event/event.js

```diff
@@ -1,17 +1,81 @@
 var eventList = [];
+var FreeEvents = [];
+var BusyEvents = [];
+const NEXT_WEEK=7*24*60*60*1000; //same time next week in mill sec
+const NEXT_SLOT=30*60*1000; //next 30 min in mill sec
 
-var Event = function(opening, recurring, startDate, endDate){
+var Event = function (opening, recurring, startDate, endDate) {
     this.opening = opening;
     this.recurring = recurring;
     this.startDate = startDate;
     this.endDate = endDate;
-
     eventList.push(this);
 };
 
-Event.prototype.availabilities = function(fromDate, toDate){
-    return //Something awesome;
+Event.prototype.availabilities = function (fromDate, toDate) {
+    eventList.forEach(function (element) {
+        if (element.opening === true) {
+            //  get free slot (start & end) in mill sec
+            start_open = Date.parse(element.startDate);
+            end_open = Date.parse(element.endDate);
+            AddFreeEvents(element, Date.parse(fromDate), Date.parse(toDate), start_open, end_open);
+        };
+    });
+    eventList.forEach(function (element) {
+        if (element.opening === false) {
+            //  get busy windows (start & end) in mill sec
+            start_busy = Date.parse(element.startDate);
+            end_busy = Date.parse(element.endDate);
+            AddBusyEvents(element, Date.parse(fromDate), Date.parse(toDate), start_busy, end_busy);
+        };
+    });
```

Comment:
Why are we looping twice here instead of looping just once and using the condition ?

Reply:
This should have be in the same loop depending on the opening state.